dotnet/src/Connectors/Connectors.Memory.CosmosDB/CosmosDBMemoryStore.cs

```diff
@@ -0,0 +1,320 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.Cosmos;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.SemanticKernel.AI.Embeddings;
+using Microsoft.SemanticKernel.AI.Embeddings.VectorOperations;
+using Microsoft.SemanticKernel.Memory;
+using Microsoft.SemanticKernel.Memory.Collections;
+using Newtonsoft.Json;
+
+namespace Connectors.Memory.CosmosDB;
+
+/// <summary>
+/// An implementation of <see cref="IMemoryStore"/> for Azure Cosmos DB.
+/// </summary>
+/// <remarks>The Embedding data is saved to the Azure Cosmos DB database container specified in the constructor.
+/// The embedding data persists between subsequent instances and has similarity search capability, handled by the client as Azure Cosmos DB is not a vector-native DB.
+/// </remarks>
+public class CosmosDBMemoryStore : IMemoryStore
+{
+    private Database _database;
+    private string _databaseName;
+    private ILogger _log;
+
+    /// <summary>
+    /// Create an instance of <see cref="CosmosDBMemoryStore"/>
+    /// </summary>
+    /// <param name="client"></param>
+    /// <param name="databaseName"></param>
+    /// <param name="log"></param>
+    /// <param name="cancel"></param>
+    /// <returns></returns>
+    /// <exception cref="CosmosException"></exception>
+    public async Task<CosmosDBMemoryStore> CreateAsync(CosmosClient client, string databaseName, ILogger? log = null, CancellationToken cancel = default)
+    {
+        this._databaseName = databaseName;
+        this._log = log ?? NullLogger<CosmosDBMemoryStore>.Instance;
+        var response = await client.CreateDatabaseIfNotExistsAsync(this._databaseName, cancellationToken: cancel);
+
+        if (response.StatusCode == HttpStatusCode.Created)
+        {
+            this._log.LogInformation("Created database {0}", this._databaseName);
+        }
+        else if (response.StatusCode == HttpStatusCode.OK)
+        {
+            this._log.LogInformation("Database {0}", this._databaseName);
+        }
+        else
+        {
+            throw new CosmosException("Database does not exist and was not created", response.StatusCode, 0, this._databaseName, 0);
```

Comment:
Is this part of `Microsoft.Azure.Cosmos`?

Reply:
Yes.